core/src/main/java/org/elasticsearch/transport/TransportChannel.java

```diff
@@ -39,4 +41,11 @@
     void sendResponse(TransportResponse response, TransportResponseOptions options) throws IOException;
 
     void sendResponse(Exception exception) throws IOException;
+
+    /**
+     * Returns the version of the other party that this channel will send a response to.
+     */
+    default Version getVersion() {
```

Comment:
can we add a test for this? Something generic in `AbstractSimpleTrasnportTestCase#testVersionFrom1to0` or so? just an assertion that we are getting the expected version